### PR TITLE
DAT-4009 Infobox Builder handle redirects when no template title set

### PR DESCRIPTION
**Pull request by dianafa**
do not merge without mercury part: https://github.com/Wikia/mercury/pull/2240

https://wikia-inc.atlassian.net/browse/DAT-4009

@SebastianMarzjan

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderHelperTest.php [hunk 1]
```diff
@@ -34,6 +34,28 @@ public function testActionSubmit( $queryStringValue, $expectedResult ) {
...
+	public function testCreateRedirectUrls( $isGood, $expected ) {
+		$statusMock = $this->getMockBuilder( 'Status' )->setMethods( [ 'isGood' ] )->getMock();
+		$statusMock->expects( $this->any() )->method( 'isGood' )->willReturn( $isGood );
+		$this->mockClass( 'Status', $statusMock );
+		$fullUrlMock = $this->getMockBuilder( 'Title' )->setMethods( [ 'getFullUrl' ] )->getMock();
+		$fullUrlMock->expects( $this->any() )->method( 'getFullUrl' )->willReturn( 'full_url' );
+		$this->mockClass( 'Title', $fullUrlMock );
+
+		$this->assertEquals( $expected, PortableInfoboxBuilderHelper::createRedirectUrls( 'test' ) );
+	}
```
Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderHelperTest.php:57]: not sure if it makes sense to mock everything here. In this way we are testing one `if`. On the other hand, only then we are UNIT testing this function independently.

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 2]
```diff
@@ -46,15 +44,35 @@ public function publish() {
...
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		$requestParams = $this->getRequest()->getParams();
+		$urls = PortableInfoboxBuilderHelper::createRedirectUrls( $requestParams['title'] );
+
+		if  ( !empty( $urls ) ) {
+			$response->setVal( 'urls', $urls );
+			$response->setVal( 'success', true );
+		} else {
+			$response->setCode( 500 );
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: Given the message I recommend using 400 Bad Request - it's the argument, not failure on the server side

File: extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ public function testTranslationFromData( $data, $expected ) {
 	 * @dataProvider markupTranslationsDataProvider
 	 */
 	public function testTranslationFromMarkup( $markup, $expected ) {
-		$generated_json = json_decode( $this->builderService->translateMarkupToData( $markup ) );
+		$generated_json = json_decode( json_encode( $this->builderService->translateMarkupToData( $markup ) ) );
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php:27]: Still missing the comment :imp:

Comment (dianafa) [inline extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php:27]: comment added where should be: https://github.com/Wikia/app/pull/10064/files#diff-682d483e4e2f7d44e7b9939661b3110eR27
:dancers:

**Comment by dianafa** [comment]
closing as this should me directed toward `DAT-3992` branch, not `dev`

**Pull request closed without merging**